diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/AbstractClassesRule.kt

```diff
@@ -46,11 +46,19 @@ class AbstractClassesRule(private val configRule: List<RulesConfig>) : Rule("abs
 
         val identifier = classNode.getFirstChildWithType(IDENTIFIER)!!.text
 
-        if (functions.isNotEmpty() && functions.none { checkAbstractModifier(it) }) {
+        if (functions.isNotEmpty() && functions.none { hasAbstractModifier(it) }) {
             CLASS_SHOULD_NOT_BE_ABSTRACT.warnAndFix(configRule, emitWarn, isFixMode, identifier, node.startOffset, node) {
-                classNode.removeChild(classNode.getFirstChildWithType(MODIFIER_LIST)!!)
-                if (classNode.firstChildNode.isWhiteSpace()) {
-                    classNode.removeChild(classNode.firstChildNode)
+                val modList = classNode.getFirstChildWithType(MODIFIER_LIST)!!
+                if (modList.getChildren(null).size > 1) {
+                    modList.removeChild(modList.getFirstChildWithType(ABSTRACT_KEYWORD)!!)
+                    if (modList.firstChildNode.isWhiteSpace()) {
```

Comment:
If `abstract` is not the first modifier, you need to remove different node with space